### PR TITLE
Added to check if the driver is invalid 

### PR DESCRIPTION
**Pull request by workdone0**
Description
Warn the user if the provided driver is not recognized, before op template resolution
Issue : https://github.com/nosqlbench/nosqlbench/issues/1006

### CODE REVIEW

**Comment by msmygit** [comment]
Thank you @workdone0 for the contribution. Will you also be adding in a unit test on this pr?

**Comment by workdone0** [comment]
@msmygit I've incorporated the [Unit Test](https://github.com/cognitree/nosqlbench/pull/8/files) into this pull request. Additionally, I've made some modifications to the test as the previous version was incorrect.

**Review by msmygit — DISMISSED** [review]
lgtm

**Review by jshook — CHANGES_REQUESTED** [review]
Just a minor structural change requested, otherwise looks good.

File: nb-engine/nb-engine-cli/src/main/java/io/nosqlbench/engine/cli/NBCLI.java [hunk 1]
```diff
@@ -218,6 +220,19 @@ public Integer applyDirect(final String[] args) {
         }
 
         final NBCLIOptions options = new NBCLIOptions(args, Mode.ParseAllOptions);
+
```
[the remark this answers is not in the document]
Comment (workdone0) [reply nb-engine/nb-engine-cli/src/main/java/io/nosqlbench/engine/cli/NBCLI.java]: @jshook I moved the logic to an adjacent class but it seems like some test are failing. I reviewed the code it looks fine, do you have any suggestions or should I revert the changes.

**Comment by jshook** [comment]
closes #1006

**Review by jshook — APPROVED** [review]
LGTM

**Review by jshook — CHANGES_REQUESTED** [review]
Let me take a closer look before merging this one

**Review by dave2wave — CHANGES_REQUESTED** [review]
Loading a driver adaptor early may lead to unexpected side effects.

This improvement should check to see if the driver adaptor exists without actually loading the driver adaptor.

**Comment by dave2wave** [comment]
There is something else going on with some adaptors due to our builds. We need to discuss (#1989) and adjust this PR accordingly.

**Review by dave2wave — APPROVED** [review]

**Comment by dave2wave** [comment]
The known, but missing adapter situation needs to be handled as a separate issue. See #2001

**Review by jshook — APPROVED** [review]
I thought there were pending issues with this, but after an other review, the changes seem specific enough not to cause concern.

**Pull request merged**